cpp/src/gandiva/precompiled/string_ops.cc

```diff
@@ -1481,6 +1481,82 @@ const char* split_part(gdv_int64 context, const char* text, gdv_int32 text_len,
   return "";
 }
 
+FORCE_INLINE
+const char* left(gdv_int64 context, const char* text, gdv_int32 text_len,
+                 gdv_int32 number, gdv_int32* out_len) {
+  // returns the 'number' left most characters of a given text
+  if (text_len == 0 || number == 0) {
+    *out_len = 0;
+    return "";
+  }
+
+  // initially counts the number of utf8 characters in the defined text
+  int32_t charCount = utf8_length(context, text, text_len);
+  // charCount is zero if input has invalid utf8 char
+  if (charCount == 0) {
```

Comment:
Better to treat the invalid bytes as single char and pad accordingly instead of returning empty string i think

Reply:
I did to keep the compatibility with the function which we based. When UTF8 char is found, an error is set.
If you think it still makes sense to enforce this behavior, I can do it